src/capi/azext_capi/custom.py

```diff
@@ -448,15 +477,44 @@ def create_workload_cluster(  # pylint: disable=unused-argument,too-many-argumen
         "AZURE_CLUSTER_IDENTITY_SECRET_NAMESPACE": os.environ["AZURE_CLUSTER_IDENTITY_SECRET_NAMESPACE"],
     }
 
+    jinja_extra_args = {
+        "EXTERNAL_CLOUD_PROVIDER": external_cloud_provider,
+        "WINDOWS": windows,
+        "EPHEMERAL": ephemeral_disks,
+        "AZURE_VNET_NAME": vnet_name,
+    }
+
     filename = capi_name + ".yaml"
     end_msg = f'✓ Generated workload cluster configuration at "{filename}"'
     with Spinner(cmd, "Generating workload cluster configuration", end_msg):
+        general_error_msg = "Could not generate workload cluster configuration."
         try:
-            manifest = template.render(args)
-            with open(filename, "w", encoding="utf-8") as manifest_file:
-                manifest_file.write(manifest)
+            if template:
+                command = ["clusterctl", "generate", "yaml", "--from"]
+                set_enviroment_variables(args)
+                if not os.path.isfile(template):
+                    domain = get_url_domain_name(template)
+                    if "github.com" not in domain:
```

Comment:
Do we need to do this special handling for GitHub? I thought if a user passed in `--template`, they were responsible for it being a complete URL pointing to a YAML template.

Reply:
It seems that the implementation on `clusterctl generate yaml` only accepts GitHub URLs and not raw template URLs. So, to handle this scenario we are producing our own request and creating a template file in case of not a github.com domain

Reply:
I was playing around with this and found this is the case:

```
clusterctl  generate yaml --from https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-azure/main/templates/addons/windows/calico/kube-proxy-windows.yaml       

Error: invalid GetFromURL operation: unable to read content from "https://raw.githubusercontent.com/kubernetes-sigs/cluster-api-provider-azure/main/templates/addons/windows/calico/kube-proxy-windows.yaml". Only reading from GitHub and local file system is supported 
```

but this works :-)

```
clusterctl  generate yaml --from https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/templates/addons/windows/calico/kube-proxy-windows.yaml 
```

Reply:
that's strange, I would expect clusterctl generate to accept raw yaml... Is there an issue in CAPI somewhere? If not, should we create one? I wonder if this is a bug or by design. If it's a bug, we shouldn't work around it.

Reply:
It seems to be by design to only accepts Github.com URLs. I open an issue for it [Add support for raw template URL in clusterctl generate yaml #6535](https://github.com/kubernetes-sigs/cluster-api/issues/6535).